### ISSUE TITLE
Groupby casts uint64 to int64, causing overflows. 

### ISSUE THREAD

**Issue by rsdenijs** [pandas-dev/pandas#25741]
#### Code Sample, a copy-pastable example if possible

```python
pd.DataFrame([['a', 12111450436133142102]],columns=['name', 'value']).groupby('name').first()
```
#### Problem description
Pandas casts uint64 to int64, causing overflows. This happens silently, causing bugs downstream. 
This happens with other groupby operations too. 
#### Expected Output
```
                    value
name                     
a    12111450436133142102
```
#### Actual Output
```
                    value
name                     
a    -6335293637576409514
```

INSTALLED VERSIONS
------------------
commit: None
python: 3.6.7.final.0
python-bits: 64
OS: Linux
OS-release: 4.4.0-135-generic
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: C.UTF-8
LANG: C.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.23.4
pytest: 3.2.2
pip: 18.1
setuptools: 40.6.3
Cython: None
numpy: 1.16.2
scipy: 1.2.1
pyarrow: None
xarray: None
IPython: 7.3.0
sphinx: None
patsy: 0.5.1
dateutil: 2.8.0
pytz: 2018.9
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: 3.0.3
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: None
sqlalchemy: 1.3.1
pymysql: None
psycopg2: 2.7.7 (dt dec pq3 ext lo64)
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
</details>

**Comment by TomAugspurger** [comment]
Can you check on 0.24.2? This seems to be fixed, at least on master.

**Comment by rsdenijs** [comment]
Correct, this bug is fixed on 0.24.2.

**Comment by TomAugspurger** [comment]
Thanks for checking.
